### PR TITLE
Add support for standalone dm integrity

### PR DESCRIPTION
**Pull request by schaefi**
There is support in kiwi to use dm_integrity in combination with the LUKS header and dm_crypt. However there is also the
use case to setup dm_integrity in standalone mode. This commit allows to create the dm_integrity layer outside of LUKS using ```/etc/integritytab``` to activate the map through a systemd generator if systemd is used.

Regarding systemd it's required to use a version of systemd which provides: ```system-generators/systemd-integritysetup-generator```. If this generator does not exist in the distribution it will also be missing in the dracut generated initrd and the boot will not be able to succeed. It's mentioned here because even newer distributions might be missing the generator

Along with the implementation there are two new optional attributes in the ```<type>``` section:

```xml

<type ... standalone_integrity="true|false" embed_integrity_metadata="true|false" .../>
```

standalone_integrity activates/deactivates the dm_integrity map on top of the root filesystem. Similar to the veritysetup support there is the opportunity to create an embedded magic metadata block at the end of the device containing the root filesystem via ```embed_integrity_metadata```

### CODE REVIEW

**Review by raimue — COMMENTED** [review]
Code looks good to me. I only added one point regarding the documentation of the dm-integrity parameters.

File: kiwi/storage/integrity_device.py [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+
+            parameters = [
+                self.integrity_algorithm
+            ] + list(integrity_superblock['flags'])
+            dm_integrity_meta = '{0} {1} {2} internal_hash:{3}'.format(
+                integrity_superblock['provided_data_sectors'],
+                integrity_superblock['sector_size'],
+                len(parameters),
+                ' '.join(parameters)
+            )
```
[the remark this answers is not in the document]
Comment (schaefi) [reply kiwi/storage/integrity_device.py]: yes good point, thanks and updated

File: doc/source/image_description/elements.rst [hunk 2]
```diff
@@ -565,6 +565,44 @@ devicepersistency="by-uuid|by-label":
...
+  dm_integrity_meta:
+    Is a string of the following information separated by spaces
+
+    * **provided_data_sectors**: number of data sectors
+    * **sector_size**: sector size in byte, defaults to 512
+    * **parameter_count**: number of `internal_hash` parameters
+    * `internal_hash`:**parameters**:
+      parameters separated by spaces, first parameter is always the
+      integrity algorithm, defaults to `sha256`. Subsequent parameters
+      are optional flags, see dm-integrity documentation on the web
```
[the remark this answers is not in the document]
Comment (schaefi) [reply doc/source/image_description/elements.rst]: yes it's misleading. Thanks for the suggestions I rephrased all this a bit and gave it a structure that allows us to add more values to internal_hash:... in the future

**Review by raimue — APPROVED** [review]
Thanks for the clarification in the documentation. Looks good to me now!

**Pull request merged**